addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/model/HmChannel.java

```diff
@@ -142,6 +136,47 @@ public boolean hasDatapoint(HmDatapointInfo dpInfo) {
         return datapoints.get(dpInfo) != null;
     }
 
+    /**
+     * Returns true, if the channel's datapoint set contains a
+     * channel function datapoint.
+     */
+    public boolean isReconfigurable() {
+        return getDatapoint(HmParamsetType.MASTER, HomematicConstants.DATAPOINT_NAME_CHANNEL_FUNCTION) != null;
+    }
+
+    public Integer getCurrentFunction() {
+        HmDatapoint functionDp = getDatapoint(HmParamsetType.MASTER,
+                HomematicConstants.DATAPOINT_NAME_CHANNEL_FUNCTION);
+        return functionDp == null ? null : (Integer) functionDp.getValue();
+    }
+
+    public boolean hasChannelFunctionChanged() {
```

Comment:
Imho this name does not suggests a side effect of storing the `lastFunction` I think this potentially a place for bugs to come from.

Reply:
Any alternative suggestions? ;-)

I initially had the comparison in the calling code and had an additional method (`acknowledgeCurrentFunction()`) that updated `lastFunction`, but that solution seemed more fragile to me because it requires the caller to do The Right Thing™. Having only one function to me seemed like the better, more simple solution.

Maybe renaming the method would be an option, something like `checkForChannelFunctionChange()`?

BTW, I should also add Javadoc comments here to document the newly added methods.

Reply:
Agree on both (name & adding java docs for it)

Reply:
I will wait with merging until you did this, agree? Can you ping me when you finished that?

Reply:
Yeah, sure ... @martinvw ping :-)